docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -864,6 +865,28 @@ https://godbolt.org/z/6qTPz9n6h
 
 ---
 
+### Vtable pointers check
+
+| Compiler Flag                                                       | Supported since     | Description                                  |
+|:--------------------------------------------------------------------|:-------------------:|:---------------------------------------------|
+| <span id="-fvtable-verify">`-fvtable-verify`</span> | GCC 4.9.4 | Enable run-time checks for virtual function pointers corruption (C++). Can impact performance. |
+
+#### Synopsis
+
+In object-oriented programming, object methods rely on dynamic dispatching to select the right virtual function to call at runtime. The vtable is the table of virtual function pointers linked to the code of the virtual function to be executed. The vtable can be overwritten by memory corruption. This option enables vtable checking at runtime for each virtual function calls. This check prevents virtual function pointers from being corrupted or overwritten, and stops the application if the pointer integrity check fails. This virtual table pointer check is based on the data structure built at application start-up.
+
+This option has three choices:
+
+- `std` - vtable checks based on data structure built after shared libraries have been loaded and initialized
+- `preinit` - vtable checks based on data structure built before shared libraries have been loaded and initialized
```

Comment:
What's the tradeoff between `std` and `preinit` and why would one choose one over the other?